ftdetect/yagpdbcc.vim

```diff
@@ -17,11 +17,27 @@
 " with this program; if not, write to the Free Software Foundation, Inc.,
 " 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-" commonly used extesions are: .go.tmpl .gotmpl .yag .yagpdb .yagcc .yagpdbcc
+" Exit early if filetypes are already loaded.
```

Comment:
This should probably go after the `yagpdbcc_set_ft` group, since we want those autocommands to be created no matter what. Currently, if the user doesn't set the override variable, no autocommands are created at all, which is a problem.

Reply:
No, this is the correct way -- it acts as a safeguard to prevent accidentally sourcing these augroups multiple times.
Creating autocommands no matter impacts performance.

If you use `:autocmd` instead of `:autocmd!`, all your detections would be defined twice. (Note that an initial `:autocmd!` cannot be used here to clear all previous definitions, because multiple `filetype.vim` files are involved, and each only adds to the existing ones.)

It is kind of similar to C's `#ifdef`.

Reply:
See also the entry in Vim's wiki https://vim.fandom.com/wiki/Filetype.vim